framework/src/timeintegrators/ExplicitRK2.C

```diff
@@ -78,6 +78,14 @@ ExplicitRK2::solve()
   _n_nonlinear_iterations += getNumNonlinearIterationsLastSolve();
   _n_linear_iterations += getNumLinearIterationsLastSolve();
 
+  // If this intermediate stage solve did not converge, then stop the entire
+  // time step solve immediately. This can be achieved by returning from this
+  // function prematurely, which guarantees that the time step convergence check
+  // will report failure. Otherwise, only the last stage would be checked for
+  // nonlinear convergence.
```

Comment:
This explains the design, so it should really be in doco for time steppers (which we do not have right now AFAIK :-( ) rather then copied into each stepper.

Reply:
I agree that it should be in time integrator documentation, but I'd also like these statements to have comments in the code because developers might find it unusual/confusing. Could I do something like the following?
```
// Abort entire time step on failure of stage - see TimeIntegrator/index.md
```

Reply:
Sure. And let's hope that `TimeIntegrator/index.md` will be around forever :-)

Reply:
Yeah... maybe just "see TimeIntegrator documentation page".